scapy/contrib/stamp.py

```diff
@@ -0,0 +1,311 @@
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        |                          Timestamp                            |
+        |                                                               |
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        |         Error Estimate        |           SSID                |
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        |                          Receive Timestamp                    |
+        |                                                               |
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        |                 Session-Sender Sequence Number                |
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        |                  Session-Sender Timestamp                     |
+        |                                                               |
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        | Session-Sender Error Estimate |           MBZ                 |
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        |Ses-Sender TTL |                   MBZ                         |
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+        ~                            TLVs                               ~
+        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    References:
+        * `Simple Two-Way Active Measurement Protocol Optional Extensions
+          [RFC 8972] <https://www.rfc-editor.org/rfc/rfc8972.html>`_
+    """
+    name = 'STAMP Session-Reflector Test'
+    fields_desc = [
+        IntField('seq', 0),
+        MultipleTypeField(
+            [
+                (TimeStampField('ts', 0),
+                    lambda pkt:pkt.err_estimate.Z == 0),
+            ],
+            UTCTimeField('ts', 0, fmt='Q')
+        ),
+        PacketField('err_estimate', ErrorEstimate(), ErrorEstimate),
+        ShortField('ssid', 1),
+        MultipleTypeField(
+            [
+                (TimeStampField('ts_rx', 0),
+                    lambda pkt:pkt.err_estimate.Z == 0)
+            ],
+            UTCTimeField('ts_rx', 0, fmt='Q')
+        ),
+        IntField('seq_sender', 0),
+        MultipleTypeField(
+            [
+                (TimeStampField('ts_sender', 0),
+                    lambda pkt:pkt.err_estimate_sender.Z == 0)
+            ],
+            UTCTimeField('ts_sender', 0, fmt='Q')
+        ),
+        PacketField('err_estimate_sender', ErrorEstimate(), ErrorEstimate),
+        ShortField('mbz1', 0),
+        ByteField('ttl_sender', 255),
+        NBytesField('mbz2', 0, 3),  # 3 bytes MBZ
+        PacketListField('tlv_objects', [], STAMPTestTLV,
+                        length_from=lambda pkt: pkt.parent.len - 8 - 44),
+    ]
+
+
+bind_layers(UDP, STAMPSessionSenderTestUnauthenticated, dport=862)
```

Comment:
You might want to also bind the source port?

Reply:
Yes, we can also bind the source port for `STAMPSessionReflectorTestUnauthenticated`.

`STAMPSessionReflectorTestUnauthenticated` is the reply to a `STAMPSessionSenderTestUnauthenticated` packet.

The Session-Sender node transmits a `STAMPSessionSenderTestUnauthenticated` to the UDP port 862 of the Session-Reflector node.
The Session-Reflector node replies by sending a `STAMPSessionReflectorTestUnauthenticated` to the Session-Sender node.

Therefore, the source port of `STAMPSessionReflectorTestUnauthenticated` is equal to the destination port of `STAMPSessionSenderTestUnauthenticated`.

We have the following mappings:
* Source port of `STAMPSessionSenderTestUnauthenticated` -> Chosen randomly
* Destination port of `STAMPSessionSenderTestUnauthenticated` -> By default, 862
* Source port of `STAMPSessionReflectorTestUnauthenticated` -> 862
* Destination port of `STAMPSessionReflectorTestUnauthenticated` -> equal to the source port of `STAMPSessionSenderTestUnauthenticated`

And we have the following bindings:
`bind_layers(UDP, STAMPSessionSenderTestUnauthenticated, dport=862)`
`bind_layers(UDP, STAMPSessionReflectorTestUnauthenticated, sport=862)`



I will update the PR to add to bind the source port. Thanks for the suggestion.